cmd/go-filecoin/daemon.go

```diff
@@ -216,18 +216,13 @@ func RunAPIAndWait(ctx context.Context, nd *node.Node, config *config.APIConfig,
 }
 
 func CreateServerEnv(ctx context.Context, nd *node.Node) *Env {
-	var storageAPI *storage.API
-	if nd.StorageProtocol != nil {
-		storageAPI = storage.NewAPI(nd.StorageProtocol.StorageClient, nd.StorageProtocol.StorageProvider, nd.PieceManager())
-	}
-
 	return &Env{
 		blockMiningAPI: nd.BlockMining.BlockMiningAPI,
 		drandAPI:       nd.DrandAPI,
 		ctx:            ctx,
 		inspectorAPI:   NewInspectorAPI(nd.Repo),
 		porcelainAPI:   nd.PorcelainAPI,
 		retrievalAPI:   nd.RetrievalProtocol,
-		storageAPI:     storageAPI,
+		storageAPI:     storage.NewAPI(nd.StorageProtocol),
```

Comment:
Thanks. Now that you've improved this we should just init and expose the StorageAPI on the node, like the other.s

Reply:
done.